CMS-Enterprise/sbom-harbor-ui
# build(storybook): enable mui theme in storybook

## Summary

This pull request incorporates updates to the Storybook setup in our `sbom-harbor-ui` project. The main focus of this PR is the wrapping of the Storybook preview in the Material UI theme provider and changes to Storybook configuration. Additionally, there are some dependency additions and updates to the TypeScript configuration.

### Added

- Wrapped Storybook preview with the Material UI theme provider. This allows the components to be rendered within Storybook with our custom Material UI themes, thus giving a more accurate preview of how the components will look in production.
- Added three new dev dependencies: `@storybook/addon-styling`, `@storybook/client-api`, and `react-docgen-typescript`. These add-ons and libraries will enhance our Storybook development environment, making it easier to work with styles and document our components.
- Changed the `.storybook/preview.ts` to a `.tsx` extension, this will allow us to write JSX in this file, which is necessary for wrapping our preview with the Material UI theme provider.

### Changed

- The `.storybook/main.ts` file has been updated to simplify how addons are added to Storybook. The new approach should be easier to maintain and expand upon in the future.
- Disabled the `noUnusedLocals` option in `compilerOptions` for `tsconfig.prod.json`. This change was necessary due to build issues with unused variables in development, but please note this change may need to be reviewed further to assess its potential impact on our production build.

### Deprecated

- n/a

### Removed

- n/a

### Fixed

- n/a

## How to test

- Pull this branch locally and run `yarn install` to ensure the new dependencies are installed correctly.
- Run `yarn storybook` and inspect the components displayed in the Storybook. They should now be displayed with the Material UI theme applied.
- Test building the application using `yarn build` and ensure there are no issues related to the TypeScript configuration change.
- Lastly, validate that all tests are still passing with `yarn test`.
